File: hotel-service/hotel-service.API/dto/response/CreateReservationResponse.cs
using hotel_service.APPLICATION.model.enums;

namespace hotel_service.APPLICATION.model.output;

public class CreateReservationResponse
{
    public string? HotelHoldId { get; set; }
    public DateTime HoldExpiry { get; set; }
    public ReservationStatus ReservationStatus { get; set; }

    private CreateReservationResponse(CreateReservationOutput output)
    {
        HotelHoldId = output.HotelHoldId;
        HoldExpiry = output.HoldExpiry;
        ReservationStatus = output.ReservationStatus;
    }

    public static CreateReservationResponse ToResponse(CreateReservationOutput output)
    {
        return new CreateReservationResponse(output);
    }
}

File: booking-service/src/main/java/com/example/bookingservice/integration/service/dto/response/CreateReservationResponse.java
package com.example.bookingservice.integration.service.dto.response;

import com.example.bookingservice.domain.model.ReservationDto;
import com.example.bookingservice.domain.model.enums.Status;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class CreateReservationResponse {
    private String hotelHoldId;
    private LocalDateTime holdExpiry;
    private Status reservationStatus;

    public ReservationDto toDto() {
        return ReservationDto.builder()
                .hotelHoldId(hotelHoldId)
                .holdExpiry(holdExpiry)
                .reservationStatus(reservationStatus)
                .build();
    }

}
